### ISSUE TITLE
Buildcraft Sealable Pipes Error

### ISSUE THREAD

**Issue by DarianLStephens** [micdoodle8/Galacticraft#2208]
The issue seems to be related to Buildcraft's Compat plugin, though I'm not sure how. Pipe ids, maybe?

When placing a sealable pipe of any type with Buildcraft's 'Compat' plugin installed, the pipe has an array of glitchy behavior, most notably it failing to function and producing the following error:
[http://pastebin.com/L0zycdaw](http://pastebin.com/L0zycdaw)
The error is from testing with a Sealable Stone Transport Pipe, though it also happens with other sealed Buildcraft pipes.

Other observed behaviors include: The pipe vanishing after placing, though only client-side (It reappears when updating the space it should be), it being deleted altogether when reloading the world, assuming you don't remove it manually, and not connecting to any other pipes.

**Comment by HadesDurin** [comment]
to be sure about your issue, can you post all of the issue report including mods list?
this part you post helps, but it is only one of the main parts of the crash report

**Comment by DarianLStephens** [comment]
Sorry, I forgot to mention that the game doesn't crash, the error appears in the log when placing a sealable Buildcraft pipe.
List of mods during testing:
Minecraft 1.7.10
Minecraft Forge 10.13.4.1558

buildcraft-7.1.16.jar
buildcraft-compat-7.1.5.jar
GalacticraftCore-1.7-3.0.12.460.jar
Galacticraft-Planets-1.7-3.0.12.460.jar
MicdoodleCore-1.7-3.0.12.460.jar

**Comment by dj3520** [comment]
I can confirm this is happening [with my server as well](http://net.dj3520.net:8000/minecraft/modlist.btm)
Now paging @asiekierka

**Comment by asiekierka** [comment]
Sounds like a Galacticraft issue to me. Is it once again not taking TileGenericPipeCompat into account?

**Comment by radfast** [comment]
OK, thanks I'll have another look at this, I thought this one was fixed already, but if it's still happening in .460 then I guess not.

**Comment by dj3520** [comment]
@radfast bumping this because I'm still seeing it happen.

**Comment by radfast** [comment]
Please try Galacticraft build 471.   That has a change to Buildcraft Sealable Pipes placement, glitchy behaviour on placement should now be fixed.

**Comment by radfast** [comment]
hmm, build 472 is still glitchy when placing Sealable pipes, if BCCompat is present.  Thank you @DarianLStephens for reporting the issue, I am able to reproduce it.

We did exactly what asiekierka said in https://github.com/micdoodle8/Galacticraft/issues/1692#issuecomment-109212081 as a "future-proof solution" :(

**Comment by asiekierka** [comment]
>Class<?> clazzBC = Class.forName("buildcraft.BuildCraftTransport");

A better route would be to just ask the GameRegistry (or Block's registry) for our block, no? The ID doesn't change...

**Comment by radfast** [comment]
Hey @asiekierka ... yes, am coding this now

**Comment by dj3520** [comment]
Question. Will the fix allow buildcraft wires to be used on a sealed variant? I have mixed feelings about them having that ability, but it makes sense that they wouldn't.

**Comment by radfast** [comment]
Found the problem.

In our code, for getting the methods needed to create and initialise the pipe, I was using .getDeclaredMethods.  But because the Block and TileEntity classes for a generic pipe can be extended by BCCompat, I needed to use .getMethods to find the superclass methods.  Not .getDeclaredMethods!  Everything else was OK.

All fixed now, thanks to @asiekierka for comments here and in #1692.
